### PR TITLE
Better doc for safe time to assume node rebooted seconds

### PR DESCRIPTION
**Pull request by mshitrit**
Related to [ECOPROJECT-1077](https://issues.redhat.com//browse/ECOPROJECT-1077)
Improving config note on `SafeTimeToAssumeNodeRebootedSeconds` in order to make it more user friendly.

### CODE REVIEW

File: api/v1alpha1/selfnoderemediationconfig_types.go
Comment (razo7) [inline api/v1alpha1/selfnoderemediationconfig_types.go:46]: missing **e** - `therefor` -> `therefore`

**Review by slintes — APPROVED** [review]

**Comment by slintes** [comment]
/hold 

let's wait if @razo7 or @beekhof have feedback on this as well

File: api/v1alpha1/selfnoderemediationconfig_types.go
Comment (mshitrit) [reply api/v1alpha1/selfnoderemediationconfig_types.go]: Thanks, fixed

File: config/crd/bases/self-node-remediation.medik8s.io_selfnoderemediationconfigs.yaml
Comment (beekhof) [inline config/crd/bases/self-node-remediation.medik8s.io_selfnoderemediationconfigs.yaml:105]: This is true and more accurate, but it's a terrible message for users.
"Set this, it might do something, it might not"

The CRD is not a substitute for exhaustive documentation, but the fact we can't concisely explain what this field does is surely a sign something needs to be improved.

Comment (mshitrit) [reply config/crd/bases/self-node-remediation.medik8s.io_selfnoderemediationconfigs.yaml:105]: >The CRD is not a substitute for exhaustive documentation

Agreed, I assume that when/after we'll merge this CR will consider best approach for ocp docs

>This is true and more accurate, but it's a terrible message for users.
"Set this, it might do something, it might not"

How about something like this ? ``Note that SafeTimeToAssumeNodeRebootedSeconds field is not meant for modifying overall remediation time ,but to verify remediation takes place once fencing is completed - therefore modifying this field will not necessarily affect the overall remediation time.``

Comment (beekhof) [reply config/crd/bases/self-node-remediation.medik8s.io_selfnoderemediationconfigs.yaml:105]: That's not really true though.  It *is* meant to control the overall remediation time, but not below a level calculated from other variables.

Comment (beekhof) [reply config/crd/bases/self-node-remediation.medik8s.io_selfnoderemediationconfigs.yaml:105]: In what way would they be different goals?

There are two times:
a) the maximum amount of time between a failure and node committing sepuku
b) the amount of time surviving peers wait before assuming the node is in a safe state (ie. dead)

They are related because while 'b' can be as long as the user wants, it must never be smaller than 'a'.

Comment (mshitrit) [reply config/crd/bases/self-node-remediation.medik8s.io_selfnoderemediationconfigs.yaml]: I've modified the comment per our conversion, feel free to edit more

**Pull request merged**
